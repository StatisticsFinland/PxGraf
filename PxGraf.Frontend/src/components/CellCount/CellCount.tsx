import { useTranslation } from 'react-i18next';
import { Alert, AlertColor } from '@mui/material';
import React from 'react';

interface ICellCountProps {
    size: number;
    maximumSize: number;
    warningLimit: number;
}

export const CellCount: React.FC<ICellCountProps> = ({ size, maximumSize, warningLimit }) => {
    const { t } = useTranslation();
    let severity: AlertColor = "info";
    let textContent: string = t("cellCount.infoText");

    if (size > maximumSize) {
        severity = "error";
        textContent = t("cellCount.dangerText");
    } else if (size >= warningLimit) {
        severity = "warning";
        textContent = t("cellCount.warningText");
    }

    return (
        <div aria-live='polite'>
            <Alert severity={severity}>{`${textContent}: ${size}/${maximumSize}`}</Alert>
        </div>
    );
}

export default CellCount;