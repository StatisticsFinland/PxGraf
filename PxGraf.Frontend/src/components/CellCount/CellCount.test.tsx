import { render } from '@testing-library/react';
import React from 'react';
import CellCount from './CellCount';

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => { }),
            },
        };
    },
}));

describe('Rendering test', () => {
    it('renders correctly with values within range', () => {
        const { asFragment } = render(<CellCount maximumSize={5} size={1} warningLimit={3} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders correctly with values over warning limit', () => {
        const { asFragment } = render(<CellCount maximumSize={5} size={4} warningLimit={3} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders correctly with values over maximum', () => {
        const { asFragment } = render(<CellCount maximumSize={5} size={6} warningLimit={3} />);
        expect(asFragment()).toMatchSnapshot();
    });
});