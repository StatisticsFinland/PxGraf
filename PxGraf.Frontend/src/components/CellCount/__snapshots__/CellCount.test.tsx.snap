// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly with values over maximum 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-3lhuho-MuiPaper-root-MuiAlert-root"
      role="alert"
      style="--Paper-shadow: none;"
    >
      <div
        class="MuiAlert-icon css-vab54s-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-zioonp-MuiAlert-message"
      >
        cellCount.dangerText: 6/5
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Rendering test renders correctly with values over warning limit 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorWarning MuiAlert-standardWarning MuiAlert-standard css-10cv5sc-MuiPaper-root-MuiAlert-root"
      role="alert"
      style="--Paper-shadow: none;"
    >
      <div
        class="MuiAlert-icon css-vab54s-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-zioonp-MuiAlert-message"
      >
        cellCount.warningText: 4/5
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Rendering test renders correctly with values within range 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-standardInfo MuiAlert-standard css-k1p3sh-MuiPaper-root-MuiAlert-root"
      role="alert"
      style="--Paper-shadow: none;"
    >
      <div
        class="MuiAlert-icon css-vab54s-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
          data-testid="InfoOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-zioonp-MuiAlert-message"
      >
        cellCount.infoText: 1/5
      </div>
    </div>
  </div>
</DocumentFragment>
`;
